deployer/src/login.ts

```diff
@@ -67,8 +67,9 @@ export async function doLogin(token: string, persister: Persister, host: string
   } else {
     debug('authorize response: %O', response)
     const auth = response.uuid + ':' + response.key
-    const credentials = await addCredentialAndSave(response.apihost, auth, response.storage, response.redis, persister, response.namespace, true)
-    persister.saveLegacyInfo(response.apihost, auth)
+    const apihost = response.apihost || host
```

Comment:
I still don't see a warning message here as was proposed in the issue.   The `||` construct will deal with the omission of apihost in the response silently, whereas it is worth saying something about it so it can get reported to us and investigated.   This is no big deal but not quite what we agreed to, I think.

Reply:
In a conversation with @bhageena we realized that in order to issue a warning here we would have to amend the signature of this function to include a Feedback object (deployer code is not allowed to write directly to the console).    This is just not worth it because a lot of things call this function (including things in the commander-cli repo) and the warning is actually only useful if observed by someone responsible for this code or by a user who chooses to report it.   Just repairing the problem silently (as in the present code) is good enough since we don't expect the error to occur in future anyway.